File: TypeScript/724. Find Pivot Index.ts
const pivotIndex = (arr: number[]): number => {
    let right: number = arr.reduce((a, c) => a + c, 0);
    let left: number = 0;

    for (let i = 0; i < arr.length; i++) {
        const current: number = arr[i]
        right -= current;

        if (left === right) {
            return i;
        }
        left += current;
    }
    return -1;
};

File: C-Sharp/724. Find Pivot Index.cs
using System.Diagnostics.CodeAnalysis;

namespace LeetCode.C_Sharp;

[SuppressMessage("ReSharper", "UnusedType.Global")]
[SuppressMessage("ReSharper", "UnusedMember.Global")]
public partial class Solution
{
    public int PivotIndex(int[] nums)
    {
        int left = 0;
        int right = 0;

        foreach (int num in nums)
        {
            right += num;
        }

        for (int i = 0; i < nums.Length; i++)
        {
            int current = nums[i];

            if (left == right - current)
            {
                return i;
            }

            right -= current;
            left += current;
        }

        return -1;
    }
}
